Sources/SwiftFormatPrettyPrint/TokenStreamCreator.swift

```diff
@@ -1497,6 +1503,40 @@ private final class TokenStreamCreator: SyntaxVisitor {
     }
   }
 
+  /// Returns `true` iff the content of the given token is immediately preceded by a newline
+  /// (ignoring whitespace and backticks).
+  private func newlinePrecedes(_ token: TokenSyntax?) -> Bool {
+    guard let token = token else { return false }
+
+    for piece in token.leadingTrivia.reversed() {
+      switch piece {
+      case .carriageReturns, .carriageReturnLineFeeds, .newlines:
+        return true
+      case .spaces, .tabs, .backticks:
+        continue
+      default:
+        return false
+      }
+    }
+    return false
+  }
+
+  /// Appends the given tokens to the stream, optionally keeping an existing newline that may be
+  /// present at the first break.
+  private func appendTokens(_ newTokens: [Token], keepExistingNewline: Bool) {
+    var convertBreakToNewline = keepExistingNewline
+    for token in newTokens {
+      if convertBreakToNewline, case .break(_, let offset) = token {
+        // To keep the existing newline, we need to change the `break` token's length to be the
+        // maximum length of the line, to force it to wrap when laid out.
+        tokens.append(.break(size: config.lineLength, offset: offset))
```

Comment:
I created member variable `self.maxlinelength`, which is set to `config.lineLength`.

Reply:
Done.